cypress/pages/BounceRulesContainer/index.js

```diff
@@ -96,65 +96,72 @@ class BounceRulesPage extends Page {
   }
 
   deleteBounceRuleAPI(testRule) {
-    return new Promise(resolve => {
-      cy.task("getRules").then(res => {
-        if (!res.errno) {
+    return cy
+      .task("getRules")
+      .then(res => {
+        if (res) {
           for (let i = 0; i < res.length; i++) {
             if (_.isEqual(testRule, _.omit(res[i], "id"))) {
-              cy.task("deleteRule", res[i].id);
-              cy.log("DELETE SUCCESS");
+              return cy.task("deleteRule", res[i].id);
             }
           }
-        } else {
-          cy.log("DELETE ERROR");
+          return true;
         }
-        cy.reload().then(() => resolve());
+        return false;
+      })
+      .then(result => {
+        if (result) {
+          cy.log("Delete Successful");
+          cy.reload();
```

Comment:
Same here with cy.reload() let's chain the overall task and open the page instead to match more what the user does